.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,77 @@
+
+- [ ] Draft [the release note](https://github.com/DataDog/dd-trace-go/releases/new) and ask the teams for its review on
+  the `#guild-dd-go` channel.
+
+- Deploy a release candidate:
+    - Create a release candidate version on this release branch:
+        - [ ] Bump and commit the release candidate version string in `/internal/version/version.go` (eg. `v1.2.3-rc.4`)
+        - [ ] Tag the bump commit with the release candidate version (eg. `git tag v1.2.3-rc.4`)
+        - [ ] Push the tag and commit (eg. `git push --tags origin release-v1.2.3`)
+    - Deploy some Datadog services to staging:
+        - [ ] Give a heads-up on the channel `#staging-headsup` that you are starting the deployment.
+        - [ ] Create a draft pull request on dd-go upgrading dd-trace-go to the release candidate tag:
+          ```console
+          dd-go/$ git checkout prod
+          dd-go/$ git pull
+          dd-go/$ git checkout -b dd-trace-go-v1.2.3
+          dd-go/$ go get -u -v gopkg.in/DataDog/dd-trace-go.v1@v1.2.3-rc.4
+          dd-go/$ go mod tidy
+          dd-go/$ git add go.mod go.sum
+          dd-go/$ git commit -m '[go.mod] upgrade dd-trace-go to v1.2.3-rc.4'
+          dd-go/$ git push -u origin dd-trace-go-v1.2.3
+          ```
+        - [ ] Deploy some services to staging:
+          ```console
+          dd-go/$ env GITLAB_TOKEN=MY_TOKEN to-staging trace-sampler \
+                                                 trace-fetcher \
+                                                 trace-firehose-writer \
+                                                 trace-stats-aggregator \
+                                                 trace-stats-extractor \
+                                                 trace-stats-query \
+                                                 fleet-config-delivery \
+                                                 runtime-security-rule-checker \
+                                                 appsec \
+                                                 fleet-edge \
+                                                 fleet-api \
+                                                 service-catalog \
+                                                 trace-edge
+          ```
+        - [ ] Check the deployment is successful on the [Go tracer dashboard]
+    - [ ] Deploy the reliability-environment services:
+        - [ ] Run [the gitlab pipeline](https://gitlab.ddbuild.io/DataDog/datadog-reliability-env/-/pipelines/new)
+          with the variable `DD_TRACE_GO_CANDIDATE_VERSION` set to the release candidate version you want to deploy.
+        - [ ] Check the deployment is successful on the [Go rel-env dashboard].
+    - [ ] Update every dd-trace-go team on `#guild-dd-go` so that they can perform any further checks.
+
+- Let the release candidate run on staging and rel-env for at least 24 hours and review all the dashboards, especially:
```

Comment:
Is this an exhaustive list? (I would try to make this as exhaustive as possible so nothing is missed)

Reply:
No, and I think we should have separate ASM, APM and Profiling validation checklist. That is a starting point to me.
cc @felixge